modules/runner/gateway_polygonbor/linux-amd64.supervisor.runner01.go

```diff
@@ -0,0 +1,438 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+type linux_amd64_supervisor_runner01_runnerdata struct {
+	Gateway         string
+	GatewayChecksum string
+}
+
+type linux_amd64_supervisor_runner01 struct {
+	Version      string
+	Storage      string
+	InstanceId   string
+	RunnerData   linux_amd64_supervisor_runner01_runnerdata
+	SkipChecksum bool
+}
+
+const (
+	runner01gatewayName               = "gateway_polygonbor_linux-amd64"
+	runner01gatewayProgramName        = "gateway_polygonbor"
+	runner01defaultUser               = "root"
+	runner01supervisorConfFiles       = "/etc/supervisor/conf.d"
+	runner01gatewaySupervisorConfFile = "gateway_polygonbor"
+	runner01logRootDir                = "/var/log/supervisor"
+	runner01oldLogRootDir             = "/var/log/old_logs"
+	runner01projectName               = "gateway_polygonbor"
+)
+
+func (r *linux_amd64_supervisor_runner01) PreRunSanity() error {
+	if !util.IsSupervisorAvailable() {
+		return errors.New("System does not support supervisor")
+	}
+	if !util.IsSupervisorInRunningState() {
+		return errors.New("System does not have supervisor in running state")
+	}
+	return nil
+}
+
+func (r *linux_amd64_supervisor_runner01) Download() error {
+	var dirPath = r.Storage + "/" + r.Version
+	err := util.CreateDirPathIfNotExists(dirPath)
+	if err != nil {
+		return err
+	}
+
+	var gatewayLocation = dirPath + "/" + runner01gatewayName
+
+	if _, err := os.Stat(gatewayLocation); os.IsNotExist(err) {
+		log.Info("Fetching gateway from upstream for version ", r.Version)
+		util.DownloadFile(gatewayLocation, r.RunnerData.Gateway)
+	}
+	if !r.SkipChecksum {
+		err := util.VerifyChecksum(gatewayLocation, r.RunnerData.GatewayChecksum)
+		if err != nil {
+			return errors.New("Error while verifying gateway checksum: " + err.Error())
+		} else {
+			log.Debug("Successully verified gateway's integrity")
+		}
+	}
+
+	err = os.Chmod(gatewayLocation, 0755)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *linux_amd64_supervisor_runner01) Prepare() error {
+	err := r.Download()
+	if err != nil {
+		return err
+	}
+
+	err = util.ChownRmarlinctlDir()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *linux_amd64_supervisor_runner01) Create(runtimeArgs map[string]string) error {
+	if _, err := os.Stat(GetResourceFileLocation(r.Storage, r.InstanceId)); err == nil {
+		return errors.New("Resource file already exisits, cannot create a new instance: " + GetResourceFileLocation(r.Storage, r.InstanceId))
+	}
+
+	currentUser, err := util.GetUser()
+	if err != nil {
+		return err
+	}
+
+	substitutions := runner01resource{
+		"linux-amd64.supervisor.runner01", r.Version, time.Now().Format(time.RFC822Z),
+		runner01gatewayProgramName + "_" + r.InstanceId, currentUser.Username, currentUser.HomeDir, r.Storage + "/" + r.Version + "/" + runner01gatewayName,
+		"", "", "", "", "", "", "",
+	}
+
+	for k, v := range runtimeArgs {
+		if k != "GatewayProgram" && k != "BridgeProgram" &&
```

Comment:
Is `BridgeProgram` referenced? No such thing in resource

Reply:
done